drone-webdriver/src/main/java/org/jboss/arquillian/drone/webdriver/binary/handler/ChromeDriverBinaryHandler.java

```diff
@@ -60,18 +67,52 @@ public String getSystemBinaryProperty() {
         return CHROME_SYSTEM_DRIVER_BINARY_PROPERTY;
     }
 
-    private class ChromeStorageSources extends XmlStorageSource {
+    public static class ChromeStorageSources extends XmlStorageSource {
 
-        ChromeStorageSources(String baseUrl) {
-            super(baseUrl, baseUrl + "LATEST_RELEASE", new HttpClient());
+        public ChromeStorageSources(String baseUrl) {
+            this(baseUrl, new HttpClient());
         }
 
+        public ChromeStorageSources(String baseUrl, HttpClient client) {
+            super(baseUrl, baseUrl + "LATEST_RELEASE", client);
+        }
+
+        @Override
+        protected String getExpectedKeyRegex(String requiredVersion, String systemVersion, String directory) {
+            return Pattern.quote(requiredVersion + "/" + getFileNameRegexToDownload(requiredVersion, systemVersion));
+        }
+
+        @Override
         protected String getExpectedKeyRegex(String requiredVersion, String directory) {
-            return Pattern.quote(requiredVersion + "/" + getFileNameRegexToDownload(requiredVersion));
+            return getExpectedKeyRegex(requiredVersion, null, directory);
+        }
+
+        @Override
+        protected ExternalBinary getLatestRelease(String charset) throws Exception {
+            ExternalBinary latestRelease;
+            try {
+                latestRelease = super.getLatestRelease(charset);
+            } catch (MissingBinaryException e) {
+                if (PlatformUtils.isWindows()) {
+                    final String latestVersion = getVersion(urlToLatestRelease, charset);
+                    log.log(Level.WARNING, "Failed downloading 64-bit version of Chrome Driver. Reason: ", e);
+                    log.log(Level.WARNING, "Downloading 32-bit version of Chrome Driver instead. ({0})", latestVersion);
+
+                    latestRelease = getReleaseForVersion(latestVersion, "32");
+                } else {
+                    throw new MissingBinaryException(e.getMessage());
+                }
+            }
+            return latestRelease;
         }
 
         @Override
         public String getFileNameRegexToDownload(String version) {
+            return getFileNameRegexToDownload(version, null);
+        }
+
+        @Override
+        public String getFileNameRegexToDownload(String version, String systemVersion) {
```

Comment:
Another reason to have something more meaningful.

I think enum `Platform` with `_32`, `_64` and `AUTO_DETECT` should work better.

Something like:

```java
public enum Platform {
    _32("32"),
    _64("64"),
    AUTO_DETECT("auto-detect"){
        @Override
        public String getPlatformCode() {
            return PlatformUtils.is32() ? "32" : "64";
        }
    };
    

    private String code;

    Platform(String code) {
        this.code = code;
    }
    
    public String getPlatformCode() {
        return this.code;
    }
}
```